extensions/emmet/src/util.ts

```diff
@@ -128,6 +128,97 @@ export function parseDocument(document: vscode.TextDocument, showError: boolean
 	return undefined;
 }
 
+export function parsePartialStylesheet(document: vscode.TextDocument, position: vscode.Position): Node | undefined {
+
+	let startPosition = new vscode.Position(0, 0);
+	let endPosition = new vscode.Position(document.lineCount - 1, document.lineAt(document.lineCount - 1).text.length);
+	const closeBrace = 125;
+	const openBrace = 123;
+	let slash = 47;
+	let star = 42;
+
+	// Go forward until we found a closing brace.
+	let stream = new DocumentStreamReader(document, position);
+	while (!stream.eof()) {
+		if (stream.eat(closeBrace)) {
+			break;
+		} else if (stream.eat(slash)) {
+			if (stream.eat(slash) && document.languageId !== 'css') {
+				// Single line Comment, we continue searching from next line.
+				stream.pos = new vscode.Position(stream.pos.line + 1, 0);
+			} else if (stream.eat(star)) {
+				// Start of block comment, we need to find the closing '*/'
+				let endCommentFound = false;
+				while (!endCommentFound) {
+					stream.eatWhile(char => { return char !== star; });
+					stream.eat(star);
+					endCommentFound = stream.eat(slash);
+				}
+			}
+		} else {
+			stream.next();
+		}
+	}
+
+	endPosition = stream.pos;
+
+	// Go back until we found an opening brace. If we find a closing one, we first find its opening brace and then we continue.
+	stream.pos = position;
+	let openBracesRemaining = 1;
+	let currentLine = position.line;
+	let isCSS = document.languageId === 'css';
+
+	while (openBracesRemaining > 0 && !stream.sof()) {
+		if (position.line - stream.pos.line > 1000) {
+			return parseStylesheet(new DocumentStreamReader(document, startPosition, new vscode.Range(startPosition, endPosition)));
+		} else if (!isCSS && stream.pos.line !== currentLine) {
```

Comment:
why check for `stream.pos.line !== currentLine` here?

Reply:
Since we are going backwards, every time we start parsing a new line we need to check if it is commented.